sklearn/metrics/pairwise.py

```diff
@@ -1324,6 +1324,13 @@ def rbf_kernel(X, Y=None, gamma=None):
     return K
 
 
+@validate_params(
+    {
+        "X": ["array-like", "sparse matrix"],
+        "Y": ["array-like", "sparse matrix", None],
+        "gamma": [Interval(Real, 0, None, closed="neither"), None],
```

Comment:
So the tests are failing because we support passing an array with a single value. It seems that this is internally intended in the Gaussian processes. We can add an hidden option then.

```suggestion
        "gamma": [
            Interval(Real, 0, None, closed="neither"),
            Hidden(np.ndarray),
            None,
        ],
```

Reply:
Thanks for helping solve this test failure!